internal/security/credentials.go

```diff
@@ -47,30 +52,81 @@ func (s *SecretProvider) GetDatabaseCredentials(database db.DatabaseInfo) (commo
 // specified path will be returned.
 func (s *SecretProvider) GetSecrets(path string, keys ...string) (map[string]string, error) {
 
+	if cachedSecrets := s.getCachedSecrets(path, keys...); cachedSecrets != nil {
+		return cachedSecrets, nil
+	}
+
+	if !s.isSecurityEnabled() {
+		s.loggingClient.Debug(fmt.Sprintf("Writable.InsecureSecrets: %v", s.configuration.Writable.InsecureSecrets))
+		return s.getInsecureSecrets(path, keys...)
+	}
+
+	newSecrets, err := s.secretClient.GetSecrets(path, keys...)
+	if err != nil {
+		return nil, err
+	}
+
+	s.secrets[path] = newSecrets
+	return s.secrets[path], nil
+}
+
+// GetInsecureSecrets retrieves secrets from the Writable section of the configuration
+// path specifies the type or location of the secrets to retrieve.
+// keys specifies the secrets which to retrieve. If no keys are provided then all the keys associated with the
+// specified path will be returned.
+func (s *SecretProvider) getInsecureSecrets(path string, keys ...string) (map[string]string, error) {
+	secrets := make(map[string]string)
+	pathExists := false
+	var missingKeys []string
+
+	for _, insecureSecrets := range s.configuration.Writable.InsecureSecrets {
+		if insecureSecrets.Path == path {
```

Comment:
1. Please guard against empty string path from configuration file as you potentially have "" for your path in the  InsecureSecrets map if not.  

2. Should we make the path case-insensitive?

Reply:
path="" is valid in call to GetSecrets() correct?

Reply:
> 1. Please guard against empty string path from configuration file as you potentially have "" for your path in the  InsecureSecrets map if not.
@jim-wang-intel `path=""` is valid in call to GetSecrets() correct?

Reply:
> 2\. Should we make the path case-insensitive?
@jim-wang-intel are paths in vault case sensitive?

Reply:
Agree with the first comment adding `&& path != ""` is a good idea

Reply:
> 
> 
> > 1. Please guard against empty string path from configuration file as you potentially have "" for your path in the  InsecureSecrets map if not.
> >    @jim-wang-intel `path=""` is valid in call to GetSecrets() correct?
Yes.

Reply:
> 
> 
> > 2. Should we make the path case-insensitive?
> > @jim-wang-intel are paths in vault case sensitive?
It seems to be that case, the path portion for Vault is case sensitive but not hostname portion.

Reply:
yes, path="" is a valid in call to GetSecrets()

Reply:
Ok, then it should be valid here.